homework27/src/ba/bitcamp/homework27/task01/FilesAndWorkerThreads.java

```diff
@@ -0,0 +1,128 @@
+package ba.bitcamp.homework27.task01;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.LinkedBlockingQueue;
+
+public class FilesAndWorkerThreads {
+
+	/**
+	 * variables
+	 */
+	private static Integer counter = 0;
+	private static BufferedReader reader = null;
+	private static char firstChar;
+	private static LinkedBlockingQueue<Runnable> jobs;
+	private static ArrayList<Producer> producers;
+	private static Object lock = new Object();
+
+	public static void main(String[] args) {
+
+		try {
+			/**
+			 * Initialising buffered reader Finding a first char in file
+			 */
+			reader = new BufferedReader(new FileReader(new File("text.txt")));
+			String lineTask = "";
+
+			firstChar = (char) reader.read();
+
+			/**
+			 * Initialising blocking queue
+			 */
+			jobs = new LinkedBlockingQueue<Runnable>();
+			while ((lineTask = reader.readLine()) != null) {
+				jobs.add(new Task(lineTask));
+			}
+
+			/**
+			 * Initialising producers Creating 8 producers(threads), staring
+			 * them and adding to array list
+			 */
+			producers = new ArrayList<>();
+
+			for (int i = 0; i < 8; i++) {
+				Producer p = new Producer();
+				p.start();
+				try {
+					p.join();
+				} catch (InterruptedException e) {
+
+					e.printStackTrace();
+				}
+				producers.add(p);
+			}
+
+			/**
+			 * Printing the result
+			 */
+			System.out.println("A character " + firstChar + " is found  "
+					+ counter + " times in file");
+
+		} catch (FileNotFoundException e) {
+
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	static class Producer extends Thread {
+
+		/**
+		 * Creating producers(threads) who takes jobs from queue, and do jobs
+		 */
+		@Override
+		public void run() {
+			while (!jobs.isEmpty()) {
+				try {
+					Runnable job = jobs.take();
+					job.run();
+				} catch (InterruptedException e) {
+					e.printStackTrace();
+				}
+			}
+
```

Comment:
Ako u ovu metodu dodam counter koja ispisuje koliko je koji producer obradio job-ova dobio bih sljedece:
Producer 0 finished. Processd 1135 jobs.
Producer 1 finished. Processd 0 jobs.
Producer 2 finished. Processd 0 jobs.
Producer 3 finished. Processd 0 jobs.
Producer 4 finished. Processd 0 jobs.
Producer 5 finished. Processd 0 jobs.
Producer 6 finished. Processd 0 jobs.
Producer 7 finished. Processd 0 jobs.
A character F is found  1859 times in file

Sta u biti ovo znaci? Ovo znaci da je prvi thread obradio sve jobove, ostali threadovi nisu nista radili. Zasto je to tako? :)